agent.go

```diff
@@ -197,6 +198,24 @@ func main() {
 	wgLoops.Wait()
 }
 
+func (p *pod) getContainer(ID string) (ctr *container) {
+	p.containersLock.RLock()
+	defer p.containersLock.RUnlock()
+
+	ctr, exist := p.containers[ID]
+	if exist == false {
+		return nil
+	}
+
+	return ctr
+}
+
+func (p *pod) setContainer(ID string, ctr *container) {
+	p.containersLock.Lock()
+	p.containers[ID] = ctr
+	p.containersLock.Unlock()
+}
+
```

Comment:
You should create a new function `deleteContainer()` locking the removal of a container:
```go
func (p *pod) deleteContainer(id string) {
    p.containersLock.Lock()
    delete(p.containers, id)
    p.containersLock.Unlock()
}
```
And use it from the code below

Reply:
Great, this has been added.